pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -0,0 +1,71 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Device Stats Monitor
+=================
+Monitors and logs device stats during training.
+"""
+from typing import Any, Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+
+class DeviceStatsMonitor(Callback):
+    r"""
+    Automatically monitors and logs device stats during training stage. ``DeviceStatsMonitor``
+    is a callback and in order to use it you need to assign a logger in the ``Trainer``.
+    Raises:
```

Comment:
the sections need to be separated by an empty line.
the docs can be built and visually inspected locally by installing the `requirements/docs.txt` and then running
```
cd docs
make html
```
the html files get generated into the docs/build directory.

Reply:
see for example here how the outputs is not rendered correctly in the current form: 

![image](https://user-images.githubusercontent.com/5495193/135174201-f1926ce1-ae94-4db9-8917-c376d7351dd8.png)

Reply:
thanks! I am getting this error trying to install requirements/docs.txt:
```
(base) daniellepintz@ip-10-200-89-146:~/repos/pytorch-lightning$ pip install -r requirements/docs.txt

ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements/../_notebooks/.actions/requirements.txt'
```

Reply:
`git submodule update --init --recursive`

probably that's missing

Reply:
thanks! got an error at first but then deleted the _notebooks directory and reran and now it works!

Reply:
now I am getting a new error when I run `make html` 😕  :

```
The name of the builder is: htmloks/course_UvA-DL/01-introduction-to-pytorch
Notebook error:
PandocMissing in notebooks/course_UvA-DL/01-introduction-to-pytorch.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
Makefile:19: recipe for target 'html' failed
make: *** [html] Error 2
(base) daniellepintz@ip-10-200-89-146:~/repos/pytorch-lightning/docs$ pip install pandoc
Requirement already satisfied: pandoc in /fsx/users/daniellepintz/conda/lib/python3.9/site-packages (1.1.0)
Requirement already satisfied: plumbum in /fsx/users/daniellepintz/conda/lib/python3.9/site-packages (from pandoc) (1.7.0)
Requirement already satisfied: ply in /fsx/users/daniellepintz/conda/lib/python3.9/site-packages (from pandoc) (3.11)
```

Reply:
hmm, I don't know about that error.
I see the formatting is now fixed so looks good now :)

Reply:
do `brew install pandoc`

Reply:
thanks Rohit! that doesn't work for me unfortunately since I have a special FB machine that has some restrictions. I will try to figure out another way

Reply:
don't worry, these docs are now fine, I checked.